### ISSUE TITLE
disable license rule for specific file

### ISSUE THREAD

**Issue by marekdef** [diffplug/spotless#650]
I have some code that has no license or code that is apache licence (like copied from a code sample).
I have also licence rule that add copyright to me, but I do not want to claim copyrights for those/

I tried
@file:Suppress("licenseHeader")
but it seems the license is still overwritten

**Comment by nedtwigg** [comment]
I think you're describing a relatively common usecase, which we don't support well.  `@file:Suppress("licenseHeader")` is a good idea, is there another tool which supports that?

There are lots of corner cases, but I think the way to support it would be to look around this string

**Comment by marekdef** [comment]
well I tried licenseHeader because `com.diffplug.spotless.generic.LicenseHeaderStep.NAME`
not that I have seen it in other tools.
We use `@file:Supress("ForbiddenImport")` and it seemed to work but this must be detekt supression

**Comment by nedtwigg** [comment]
This isn't a good solution, but you can now disable spotless formatting with [`spotless:off` and `spotless:on` tags](https://github.com/diffplug/spotless/tree/main/plugin-gradle#spotlessoff-and-spotlesson).  If you make your license header look like this:

```
/**
 * Blah blah
 * Blah blah blah blah
 *
 * Licensing exceptions below:
 * spotless:off (none) spotless:on
 */
```

Then if you replace the stuff inside of `(none)`, including multiple lines, that stuff will be preserved.  That's not a good solution, but it's the best we have for now.

**Comment by nedtwigg** [comment]
Published in plugin-gradle `6.0.1`. Currently unsupported in maven, a PR for that is welcome.
